examples/platforms/efr32mg12/efr32mg12.ld

```diff
@@ -0,0 +1,241 @@
+/***************************************************************************//**
+ * Linker script for Silicon Labs EFR32MG12P devices
+ * @version 5.8.1
+ *******************************************************************************
+ * # License
+ * <b>Copyright 2019 Silicon Laboratories Inc. www.silabs.com</b>
```

Comment:
Can we move this to somewhere in `third_party/silabs`? All files that carry a license other than the standard OpenThread license header need to be placed in `third_party`.

Reply:
The linker files are different for mg12,mg13,mg21 etc. so the logical place to put them is on the individual platforms/efr32mgxx directories - we can change the file license header to an 'openThread'  type as necessary (there's nothing in the linker scripts which is Silabs proprietory info)

Reply:
Providing this file with the standard OpenThread license header would work if you want to keep it out of `third_party/`.